mesa/datacollection.py

```diff
@@ -244,6 +246,7 @@ def get_agent_vars_dataframe(self):
             columns=["Step", "AgentID", *rep_names],
             index=["Step", "AgentID"],
         )
+        df.index.names = ["Step", "AgentID"]
```

Comment:
Please see above, same concern with parameter sweeps.

Reply:
Apologies for the multiple message had some extra time free up so was able to look more deeply  --- why do you need to add line 249; line 247 makes the index "Step" and "AgentID"